compiler/rustc_middle/src/mir/visit.rs

```diff
@@ -367,10 +367,18 @@ macro_rules! make_mir_visitor {
                     ) => {
                         self.visit_assign(place, rvalue, location);
                     }
-                    StatementKind::FakeRead(box (_, place)) => {
+                    StatementKind::FakeRead(box (cause, place)) => {
+                        let context = match cause {
+                            FakeReadCause::ForWildcard => {
+                                PlaceContext::NonUse(NonUseContext::ForWildcard)
```

Comment:
I think this updated PR does semantically what I was hoping, but this particular part emphasizes a bit of uncomfort I have with the specific MIR form of it.  Needing a check for whether a "read" should be considered a read seems odd to me.  How people people feel about non-use cases being in a different statementkind?  [On zulip](https://rust-lang.zulipchat.com/#narrow/stream/213817-t-lang/topic/underscoring.20meaning.20of.20underscore/near/303716070) I was using `PlaceMention` as a strawman for this, and borrowck would then treat that as a `NonUse`, rather than having a `FakeRead` be sometimes a use a sometimes not a use.

Reply:
I was considering this above, but I'm somewhat unconvinced of the value of it while we don't really know what `FakeRead` is, especially from an opsem perspective. We should definitely fix this eventually but doing it once when we're sure feels like enough